### PR TITLE
Add nuget signing check

### PR DESCRIPTION
**Pull request by wli3**
Verify after package download

### CODE REVIEW

File: src/Cli/dotnet/NugetPackageDownloader/FirstPartyNuGetPackageSigningVerifier.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            }
+            catch (FileNotFoundException)
+            {
+                return false;
+            }
+        }
+
+        private static bool NuGetVerify(FilePath nupkgToVerify, out string commandOutput)
+        {
+            var argsToPassToRestore = new[] {"verify", "--all", nupkgToVerify.Value};
```
Comment (sfoslund) [inline src/Cli/dotnet/NugetPackageDownloader/FirstPartyNuGetPackageSigningVerifier.cs:49]: nit: argsToPassToNuget?

File: src/Cli/dotnet/NugetPackageDownloader/NuGetPackageDownloader.cs [hunk 2]
```diff
@@ -28,13 +32,26 @@ internal class NuGetPackageDownloader : INuGetPackageDownloader
         };
 
         private readonly IFilePermissionSetter _filePermissionSetter;
-        private readonly ILogger _logger;
+        
+        /// <summary>
+        /// In many commands we don't passing console logger to reduce the verbosity
```
Comment (sfoslund) [inline src/Cli/dotnet/NugetPackageDownloader/NuGetPackageDownloader.cs:37]: I don't understand this comment, can you clarify/ reword?

File: src/Cli/dotnet/NugetPackageDownloader/NuGetPackageDownloader.cs [hunk 3]
```diff
@@ -70,18 +87,56 @@ public NuGetPackageDownloader(DirectoryPath packageInstallDir, IFilePermissionSe
...
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                if (_firstPartyNuGetPackageSigningVerifier.IsExecutableIsFirstPartySignedWithoutValidation(new FilePath(
+                    typeof(DotNet.Cli.Program).Assembly.Location)))
+                {
+                    if (!_firstPartyNuGetPackageSigningVerifier.Verify(new FilePath(nupkgPath),
+                        out string commandOutput))
+                    {
+                        throw new NuGetPackageInstallerException(LocalizableStrings.FailedToValidatePackageSigning +
```
Comment (sfoslund) [inline src/Cli/dotnet/NugetPackageDownloader/NuGetPackageDownloader.cs:114]: Do we just leave the unsigned package on the machine in this case? Is that a problem, should we delete it/ clean up after ourselves, especially if it's unsigned?

File: src/Cli/dotnet/commands/dotnet-workload/repair/WorkloadRepairCommand.cs [hunk 4]
```diff
@@ -57,7 +57,8 @@ internal class WorkloadRepairCommand : CommandBase
                 Path.GetTempPath() :
                 parseResult.ValueForOption<string>(WorkloadInstallCommandParser.TempDirOption));
             var tempPackagesDir = new DirectoryPath(Path.Combine(tempDirPath, "dotnet-sdk-advertising-temp"));
-            nugetPackageDownloader = nugetPackageDownloader ?? new NuGetPackageDownloader(tempPackagesDir, filePermissionSetter: null, new NullLogger());
+            NullLogger nullLogger = new NullLogger();
+            nugetPackageDownloader ??= new NuGetPackageDownloader(tempPackagesDir, filePermissionSetter: null, new FirstPartyNuGetPackageSigningVerifier(tempPackagesDir, nullLogger),nullLogger);
```
Comment (sfoslund) [inline src/Cli/dotnet/commands/dotnet-workload/repair/WorkloadRepairCommand.cs:61]: nit: space between `...,nullLogger`

File: src/Cli/dotnet/commands/dotnet-workload/uninstall/WorkloadUninstallCommand.cs [hunk 5]
```diff
@@ -42,8 +42,8 @@ internal class WorkloadUninstallCommand : CommandBase
             _sdkVersion = WorkloadOptionsExtensions.GetValidatedSdkVersion(parseResult.ValueForOption<string>(WorkloadUninstallCommandParser.VersionOption), version, dotnetPath);
             var verbosity = parseResult.ValueForOption<VerbosityOptions>(WorkloadUninstallCommandParser.VerbosityOption);
             var workloadManifestProvider = new SdkDirectoryWorkloadManifestProvider(dotnetPath, _sdkVersion.ToString());
-            workloadResolver = workloadResolver ?? WorkloadResolver.Create(workloadManifestProvider, dotnetPath, _sdkVersion.ToString());
-            nugetPackageDownloader = nugetPackageDownloader ?? new NuGetPackageDownloader(new DirectoryPath(Path.GetTempPath()), filePermissionSetter: null, new NullLogger());
+            workloadResolver ??= WorkloadResolver.Create(workloadManifestProvider, dotnetPath, _sdkVersion.ToString());
+            nugetPackageDownloader ??= new NuGetPackageDownloader(new DirectoryPath(Path.GetTempPath()), filePermissionSetter: null, verboseLogger: new NullLogger());
```
Comment (sfoslund) [inline src/Cli/dotnet/commands/dotnet-workload/uninstall/WorkloadUninstallCommand.cs:46]: 👍

File: src/Tests/Microsoft.DotNet.PackageInstall.Tests/NuGetPackageInstallerTests.cs [hunk 6]
```diff
@@ -154,6 +156,71 @@ public async Task WhenPassedIncludePreviewItInstallSucceeds()
...
+            NuGetPackageInstallerException ex = await Assert.ThrowsAsync<NuGetPackageInstallerException>(() =>
+                nuGetPackageDownloader.DownloadPackageAsync(
+                    TestPackageId,
+                    new NuGetVersion(TestPackageVersion),
+                    new PackageSourceLocation(sourceFeedOverrides: new[] {GetTestLocalFeedPath()})));
+
+            ex.Message.Should().Contain(commandOutput);
+        }
+
+        [PlatformSpecificFact(TestPlatforms.Linux | TestPlatforms.OSX | TestPlatforms.FreeBSD)]
```
Comment (sfoslund) [inline src/Tests/Microsoft.DotNet.PackageInstall.Tests/NuGetPackageInstallerTests.cs:200]: UnixOnlyFact?

**Review by sfoslund — APPROVED** [review]

File: src/Cli/dotnet/NugetPackageDownloader/NuGetPackageDownloader.cs (hunk 2 above)
Comment (wli3) [reply src/Cli/dotnet/NugetPackageDownloader/NuGetPackageDownloader.cs:37]: "In many commands we don't passing console logger and pass NullLogger instead to reduce the verbosity"

Comment (wli3) [reply src/Cli/dotnet/NugetPackageDownloader/NuGetPackageDownloader.cs:37]: will that clarify? We pass null logger, and so there is mostly nothing printed to the screen for a lot of commands like list and install

File: src/Cli/dotnet/NugetPackageDownloader/NuGetPackageDownloader.cs (hunk 3 above)
Comment (wli3) [reply src/Cli/dotnet/NugetPackageDownloader/NuGetPackageDownloader.cs:114]: good point. We should delete it

**Pull request merged**